hail/src/main/scala/is/hail/expr/ir/ExtractAggregators.scala

```diff
@@ -142,13 +142,32 @@ object ExtractAggregators {
                 ArrayRange(I32(0), ArrayLen(Ref(aUID, a.typ)), I32(1)),
                 iUID,
                 Let(
-                  name,
+                  elementName,
                   ArrayRef(Ref(aUID, a.typ), Ref(iUID, TInt32())),
-                  SeqOp(
+                  addLets(SeqOp(
```

Comment:
I'm having trouble understanding why the `addLets` got moved here.

Reply:
This was a bug - the lets added here are ones that mention the `elementName` binding, so they need to go inside the binding!

This fix was required to pass tests.